### ISSUE TITLE
azure network dns record-set delete-record has issue

### ISSUE THREAD

**Issue by elemantalcode** [Azure/azure-xplat-cli#3058]
CLI Version: **0.10.2**
OS Type: CentOS 6.8 and CentOS 7.2
Installation via:  npm (also tried git)

Mode: **ARM**

Environment: **AzureCloud**

Description:
Unable to remove an A record. DNS zone is created and working. Unable to remove record.

Example DNS:

`azure network dns record-set list BaseResourceGroup domain.co.uk` 

Result: 

```
info:    Executing command network dns record-set list
+ Looking up the DNS Record Sets
data:    Name                            TTL     Type  Records                                     Tags
data:    ------------------------------  ------  ----  ------------------------------------------  ----
data:    @                               172800  NS    ns1-05.azure-dns.com.
data:                                                  ns2-05.azure-dns.net.
data:                                                  ns3-05.azure-dns.org.
data:                                                  ns4-05.azure-dns.info.
data:    @                               3600    SOA   host:    ns1-05.azure-dns.com.
data:                                                  email:   azuredns-hostmaster.microsoft.com
data:                                                  refresh: 3600
data:                                                  retry:   300
data:                                                  expire:  2419200
data:                                                  minimum: 300
data:    host1                         300     A     172.17.5.4
data:    host2                         300     A     172.17.6.4
data:    host3                         300     A     172.17.6.8
data:    host4                         300     A     172.17.6.6
data:    host5                         300     A     172.17.6.5
data:    host6                         300     A     172.17.6.7
```

Try to remove:

`azure network dns record-set delete-record --resource-group BaseResourceGroup --dns-zone-name domain.co.uk --record-set-name host4 --type A --ipv4-address 172.17.6.6`

Result:

```
info:    Executing command network dns record-set delete-record
+ Looking up the DNS Record Set "host4" of type "A"
Delete DNS Record? [y/n]  y
+ Updating record set "host4"
error:   parameters.recordSet.location cannot be null.
error:   Error information has been recorded to /home/azure/.azure/azure.err
error:   network dns record-set delete-record command failed
```

Error stack trace:

```
2016-08-01T12:20:42.832Z:
{ [Error: parameters.recordSet.location cannot be null.]
  stack: [Getter/Setter],
  __frame:
   { name: 'updateRecordSet__20',
     line: 479,
     file: '/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js',
     prev:
      { name: 'deleteRecord__19',
        line: 460,
        file: '/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js',
        prev: [Object],
        calls: 2,
        active: false,
        offset: 15,
        col: 23 },
     calls: 0,
     active: false,
     offset: 4,
     col: 58 },
  rawStack: [Getter] }
Error: parameters.recordSet.location cannot be null.
  <<< async stack >>>
    at RecordSetOperations.createOrUpdate (/usr/lib/node_modules/azure-cli/node_modules/azure-arm-dns/lib/dnsManagementClient.js:238:23)
    at updateRecordSet__20 (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js:483:62)
  at updateRecordSet__20 (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js:483:59)
  at deleteRecord__19 (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js:475:24)
  at __82 (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/network.js:1742:15)
  <<< raw stack >>>
    at RecordSetOperations.createOrUpdate (/usr/lib/node_modules/azure-cli/node_modules/azure-arm-dns/lib/dnsManagementClient.js:238:23)
    at __$updateRecordSet__20 (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js:483:62)
    at __tryCatch (/usr/lib/node_modules/azure-cli/node_modules/streamline/lib/callbacks/runtime.js:150:4)
    at ___ (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js:481:250)
    at ___ (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js:484:65)
    at __$updateRecordSet__20 (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js:489:79)
    at __tryCatch (/usr/lib/node_modules/azure-cli/node_modules/streamline/lib/callbacks/runtime.js:150:4)
    at ___ (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js:481:150)
    at ___ (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js:489:182)
    at __$updateRecordSet__20 (/usr/lib/node_modules/azure-cli/lib/commands/arm/network/dnsZone.js:491:159)
```

Seems to be linked to a similar problem with adding records found here: #2989

**Comment by MikhailTryakhov** [comment]
Hi @elemantalcode 
could you please make sure that your azure-cli package version is really **0.10.2**.
It was fixed in previous version and now it's non-reproducable in specified 0.10.2 version

**Comment by elemantalcode** [comment]
Hi @MikhailTryakhov 

Yes. It's 0.10.2:

```
[azure@pc6-8 ~]$ /usr/bin/azure -v
0.10.2
```

**Comment by MikhailTryakhov** [comment]
Strange, still can't reproduce it.
Added workaround to prevent specified requirement issues - would be published in release 0.10.3.
Please let me know if you would have any problems

**Comment by elemantalcode** [comment]
Hi @MikhailTryakhov,

I've upgraded to 0.10.6. I can confirm that this works fine now, thank you.
